sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqMessage.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.rabbitmq;
+
+import com.rabbitmq.client.AMQP;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/** It contains the message payload, and additional metadata like routing key or attributes. */
+public class RabbitMqMessage implements Serializable {
+  private final String routingKey;
+  private final byte[] body;
+  private final String contentType;
+  private final String contentEncoding;
+  private final Map<String, Object> headers;
+  private final Integer deliveryMode;
+  private final Integer priority;
+  private final String correlationId;
+  private final String replyTo;
+  private final String expiration;
+  private final String messageId;
+  private final Date timestamp;
+  private final String type;
+  private final String userId;
+  private final String appId;
+  private final String clusterId;
+
+  public RabbitMqMessage(byte[] body) {
+    this.body = body;
+    routingKey = "";
+    contentType = null;
+    contentEncoding = null;
+    headers = new HashMap<>();
+    deliveryMode = 1;
+    priority = 1;
+    correlationId = null;
+    replyTo = null;
+    expiration = null;
+    messageId = null;
```

Comment:
Per implementation of equals() below it seems that two default-constructed messages with the same body will be non-equal. Is this expected?